ldap-client_test.go

```diff
@@ -80,46 +100,23 @@ func Test_LDAPClient(t *testing.T) {
 
 	t.Run("GetAllGroupsByName", func(t *testing.T) {
 		client := &LDAPClient{
-			Base:         "ou=People,dc=planetexpress,dc=com",
-			Host:         "127.0.0.1",
-			Port:         389,
+			Base:        "dc=planetexpress,dc=com",
+			Host:        "localhost",
+			Port:        10389,
 			UseSSL:       false,
 			BindDN:       "cn=admin,dc=planetexpress,dc=com",
 			BindPassword: "GoodNewsEveryone",
 			UserFilter:   "(uid=%s)",
 			GroupFilter:  "(memberUid=%s)",
-			Attributes:   []string{"givenName", "sn", "mail", "uid"},
+			Attributes:   []string{"cn", "member", "memberUid"},
 		}
 		defer client.Close()
 
-		results, err := client.GetAllGroupsByName("ship_crew")
+		results, err := client.GetAllGroupsByName("_")
 		if err != nil {
 			log.Fatalf("Error getting all groups  matching to %s, %v", "ship_crew", err)
 		}
 
 		log.Printf("found: %+v", results)
 	})
-
-	t.Run("ChangeUserPassword", func(t *testing.T) {
```

Comment:
Let's re-add it, and call on the client "ImplementationType" in order to know if to call "ChangeADUserPassword" or "ChangeOpenLDAPUserPassword".